Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -6,8 +6,155 @@
  *
  */
 
-#include <AzCore/Memory/AllocatorManager.h>
 #include <AzCore/Memory/Memory.h>
+#include <AzCore/Memory/AllocatorManager.h>
+
+#define RECORDING_ENABLED 0
+
+#if RECORDING_ENABLED
+
+#include <AzCore/std/containers/map.h>
+#include <AzCore/IO/SystemFile.h>
+#include <AzCore/std/parallel/mutex.h>
+#include <AzCore/std/parallel/scoped_lock.h>
+
+namespace
+{
+    class DebugAllocator
+    {
+    public:
+        typedef void* pointer_type;
+        typedef AZStd::size_t size_type;
+        typedef AZStd::ptrdiff_t difference_type;
+        typedef AZStd::false_type allow_memory_leaks; ///< Regular allocators should not leak.
+
+        AZ_FORCE_INLINE pointer_type allocate(size_t byteSize, size_t alignment, int = 0)
+        {
+            return AZ_OS_MALLOC(byteSize, alignment);
+        }
+        AZ_FORCE_INLINE size_type resize(pointer_type, size_type)
+        {
+            return 0;
+        }
+        AZ_FORCE_INLINE void deallocate(pointer_type ptr, size_type, size_type)
+        {
+            AZ_OS_FREE(ptr);
+        }
+    };
+
+    #pragma pack(push, 1)
```

Comment:
Is this pragma actually needed with the `alignas` attribute below?

Reply:
yes, this struct is used to deserialize data from a file. In order to make the file smaller, I packed it to reduce size. The file is a binary-serialized array of this struct.